### ISSUE TITLE
[Bug]: Load collection failed after compact and create index

### ISSUE THREAD

**Issue by ThreadDao** [milvus-io/milvus#12146]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Environment

```markdown
- Milvus version: master-20211118-27715c1
- Deployment mode(standalone or cluster): cluster
- SDK version(e.g. pymilvus v2.0.0rc2): pymilvus-2.0.0rc9.dev11
- OS(Ubuntu or CentOS): 
- CPU/Memory: 
- GPU: 
- Others: enableCompaction: true
          compaction.retentionDuration: 20
```

### Current Behavior

1. Create with `shard_num=1`
2. Insert 3000 entities, flush
3. Insert 3000 entities, flush
4. compact
5. create index `default_index = {"index_type": "IVF_SQ8", "metric_type": "L2", "params": {"nlist": 64}}`
6. load
Load collection return exception:
```
[2021-11-19 20:02:34,038 - DEBUG - ci_test]: (api_response) : {"name": "compact_zmutG4tA", "schema": {"auto_id": false, "description": "", "fields": [{"name": "int64", "description": "", "type": 5, "is_primary": true, "auto_id": false}, {"name": "float", "description": "", "type": 10}, {"name": "float_vector", "description": "", "type": 101, "params": {"dim": ......  (api_request.py:21)
[2021-11-19 20:02:34,160 - DEBUG - ci_test]: (api_request)  : [Collection.insert] args: [      int64   float                                       float_vector
0         0     0.0  [0.1459192792406115, 0.05086554267027452, 0.12...
1         1     1.0  [0.022639014805284383, 0.1477245583655668, 0.1...
2         2     2.0  [0.07348938954198107, 0.09126849403434108, 0.0...
3         3    ......, kwargs: {'timeout': 20} (api_request.py:44)
[2021-11-19 20:02:48,334 - DEBUG - ci_test]: (api_response) : (insert count: 3000, delete count: 0, upsert count: 0, timestamp: 429214497027129346)  (api_request.py:21)
[2021-11-19 20:02:51,615 - DEBUG - ci_test]: (api_request)  : [Collection.insert] args: [      int64   float                                       float_vector
0         0     0.0  [0.07977108731270963, 0.004874807289357237, 0....
1         1     1.0  [0.018738438907820086, 0.0669336112512885, 0.1...
2         2     2.0  [0.002535506951402409, 0.08492849613622214, 0....
3         3    ......, kwargs: {'timeout': 20} (api_request.py:44)
[2021-11-19 20:03:05,494 - DEBUG - ci_test]: (api_response) : (insert count: 3000, delete count: 0, upsert count: 0, timestamp: 429214501536006145)  (api_request.py:21)
[2021-11-19 20:03:08,629 - DEBUG - ci_test]: (api_request)  : [Collection.compact] args: [20], kwargs: {} (api_request.py:44)
[2021-11-19 20:03:08,650 - DEBUG - ci_test]: (api_response) : None  (api_request.py:21)
[2021-11-19 20:03:09,673 - DEBUG - ci_test]: True (collection_wrapper.py:263)
[2021-11-19 20:03:09,674 - DEBUG - ci_test]: (api_request)  : [Collection.get_compaction_plans] args: [20], kwargs: {} (api_request.py:44)
[2021-11-19 20:03:09,683 - DEBUG - ci_test]: (api_response) : 
Compaction Plans:
 - compaction id: 429214502387974145
 - state: 2
 - plans: [
Plan:
 - sources: [429214497040236545, 429214501549113345]
 - target: 429214502387974147
]
   (api_request.py:21)
[2021-11-19 20:03:09,684 - DEBUG - ci_test]: (api_request)  : [Collection.create_index] args: ['float_vector', {'index_type': 'IVF_SQ8', 'metric_type': 'L2', 'params': {'nlist': 64}}], kwargs: {'timeout': 20} (api_request.py:44)
[2021-11-19 20:03:13,896 - DEBUG - ci_test]: (api_response) : Status(code=0, message='')  (api_request.py:21)
[2021-11-19 20:03:13,896 - DEBUG - ci_test]: (api_request)  : [Collection.index] args: [], kwargs: {} (api_request.py:44)
[2021-11-19 20:03:13,907 - DEBUG - ci_test]: (api_response) : <pymilvus.orm.index.Index object at 0x7f52483415b0>  (api_request.py:21)
[2021-11-19 20:03:13,907 - DEBUG - ci_test]: (<pymilvus.orm.index.Index object at 0x7f52483415b0>, True) (test_compaction.py:290)
[2021-11-19 20:03:13,908 - DEBUG - ci_test]: (api_request)  : [Collection.load] args: [None], kwargs: {'timeout': 20} (api_request.py:44)
[2021-11-19 20:03:18,637 - ERROR - ci_test]: Traceback (most recent call last):
  File "/home/zong/zong/projects/milvus/tests/python_client/utils/api_request.py", line 18, in inner_wrapper
    res = func(*args, **kwargs)
  File "/home/zong/zong/projects/milvus/tests/python_client/utils/api_request.py", line 45, in api_request
    return func(*arg, **kwargs)
  File "/home/zong/zong/.virtualenvs/milvus/lib/python3.8/site-packages/pymilvus/orm/collection.py", line 449, in load
    conn.load_collection(self._name, timeout=timeout, **kwargs)
  File "/home/zong/zong/.virtualenvs/milvus/lib/python3.8/site-packages/pymilvus/client/stub.py", line 58, in handler
    raise e
  File "/home/zong/zong/.virtualenvs/milvus/lib/python3.8/site-packages/pymilvus/client/stub.py", line 42, in handler
    return func(self, *args, **kwargs)
  File "/home/zong/zong/.virtualenvs/milvus/lib/python3.8/site-packages/pymilvus/client/stub.py", line 322, in load_collection
    return handler.load_collection("", collection_name=collection_name, timeout=timeout, **kwargs)
  File "/home/zong/zong/.virtualenvs/milvus/lib/python3.8/site-packages/pymilvus/client/grpc_handler.py", line 75, in handler
    raise e
  File "/home/zong/zong/.virtualenvs/milvus/lib/python3.8/site-packages/pymilvus/client/grpc_handler.py", line 67, in handler
    return func(self, *args, **kwargs)
  File "/home/zong/zong/.virtualenvs/milvus/lib/python3.8/site-packages/pymilvus/client/grpc_handler.py", line 823, in load_collection
    self.wait_for_loading_collection(collection_name, timeout)
  File "/home/zong/zong/.virtualenvs/milvus/lib/python3.8/site-packages/pymilvus/client/grpc_handler.py", line 75, in handler
    raise e
  File "/home/zong/zong/.virtualenvs/milvus/lib/python3.8/site-packages/pymilvus/client/grpc_handler.py", line 67, in handler
    return func(self, *args, **kwargs)
  File "/home/zong/zong/.virtualenvs/milvus/lib/python3.8/site-packages/pymilvus/client/grpc_handler.py", line 841, in wait_for_loading_collection
    return self._wait_for_loading_collection_v2(collection_name, timeout)
  File "/home/zong/zong/.virtualenvs/milvus/lib/python3.8/site-packages/pymilvus/client/grpc_handler.py", line 868, in _wait_for_loading_collection_v2
    raise BaseException(response.status.error_code, response.status.reason)
pymilvus.client.exceptions.BaseException: <BaseException: (code=1, message=rpc error: code = Unknown desc = collection 429214493265562945 has not been loaded to memory or load failed)>
 (api_request.py:26)
[2021-11-19 20:03:18,638 - ERROR - ci_test]: (api_response) : <BaseException: (code=1, message=rpc error: code = Unknown desc = collection 429214493265562945 has not been loaded to memory or load failed)> (api_request.py:27)
FAILED
```
Server logs:
[milvus_logs.tar.gz](https://github.com/milvus-io/milvus/files/7570146/milvus_logs.tar.gz)
 

### Expected Behavior

Load successfully

### Steps To Reproduce

```markdown
def test_compact_and_index(self):
        """
        target: test compact after create index
        method: 1.insert data into two segments
                2.create index
                3.compact
                4.search
        expected: Verify segment info and index info
        """
        collection_w = self.collection_insert_multi_segments_one_shard(prefix, nb_of_segment=ct.default_nb)

        # compact
        collection_w.compact()
        collection_w.wait_for_compaction_completed()
        collection_w.get_compaction_plans()

        # create index
        collection_w.create_index(ct.default_float_vec_field_name, ct.default_index)
        log.debug(collection_w.index())

        # search
        vectors = [[random.random() for _ in range(ct.default_dim)] for _ in range(ct.default_nq)]
        collection_w.load()
        search_res, _ = collection_w.search(vectors,
                                            ct.default_float_vec_field_name,
                                            ct.default_search_params, ct.default_limit,
                                            check_task=CheckTasks.check_search_results,
                                            check_items={"nq": ct.default_nq,
                                                         "ids": [],
                                                         "limit": 0}
                                            )
```
```


### Anything else?

_No response_

**Comment by xige-16** [comment]
[2021/11/19 12:03:14.621 +00:00] [DEBUG] [cluster.go:263] ["loadSegments: queryNode load segments error"] [nodeID=7] ["error info"="rpc error: code = Unknown desc = [UnexpectedError] load data has different row count from other columns"]

**Comment by xige-16** [comment]
It seems that the binlog is written incorrectly, or the insert log file path of some columns is missing during load.

**Comment by xige-16** [comment]
1B benchmarck test has reproduced the problem
            NOMINATED NODE   READINESS GATES
benchmark-test-7wpf9-1-etcd-0                                 1/1     Running     0          3m3s    10.97.17.66    qa-node014.zil
liz.local   <none>           <none>
benchmark-test-7wpf9-1-etcd-1                                 1/1     Running     0          3m3s    10.97.17.64    qa-node014.zil
liz.local   <none>           <none>
benchmark-test-7wpf9-1-etcd-2                                 1/1     Running     0          3m3s    10.97.17.67    qa-node014.zil
liz.local   <none>           <none>
benchmark-test-7wpf9-1-milvus-datacoord-585997c675-795xw      1/1     Running     0          3m3s    10.97.12.42    qa-node015.zil
liz.local   <none>           <none>
benchmark-test-7wpf9-1-milvus-datanode-ccff88978-sn7dv        1/1     Running     0          3m2s    10.97.17.63    qa-node014.zil
liz.local   <none>           <none>
benchmark-test-7wpf9-1-milvus-datanode-ccff88978-w26xp        1/1     Running     0          3m2s    10.97.12.43    qa-node015.zil
liz.local   <none>           <none>
benchmark-test-7wpf9-1-milvus-indexcoord-5cd49df484-xs9p5     1/1     Running     0          3m3s    10.97.19.83    qa-node016.zil
liz.local   <none>           <none>
benchmark-test-7wpf9-1-milvus-indexnode-5c6674dd5c-dmx5t      1/1     Running     0          3m2s    10.97.15.165   qa-node012.zil
liz.local   <none>           <none>
benchmark-test-7wpf9-1-milvus-indexnode-5c6674dd5c-gb59k      1/1     Running     0          3m2s    10.97.11.108   qa-node009.zil
liz.local   <none>           <none>
benchmark-test-7wpf9-1-milvus-indexnode-5c6674dd5c-gbf6m      1/1     Running     0          3m2s    10.97.9.136    qa-node007.zil
liz.local   <none>           <none>
benchmark-test-7wpf9-1-milvus-indexnode-5c6674dd5c-qnb82      1/1     Running     0          3m2s    10.97.19.85    qa-node016.zil
liz.local   <none>           <none>
benchmark-test-7wpf9-1-milvus-proxy-599dbfc5b5-q42jh          1/1     Running     0          3m3s    10.97.11.106   qa-node009.zil
liz.local   <none>           <none>
benchmark-test-7wpf9-1-milvus-querycoord-77fbfdbff5-lnhnz     1/1     Running     0          3m3s    10.97.12.41    qa-node015.zil
liz.local   <none>           <none>
benchmark-test-7wpf9-1-milvus-querynode-598b88956c-c2t6w      1/1     Running     0          3m3s    10.97.15.164   qa-node012.zil
liz.local   <none>           <none>
benchmark-test-7wpf9-1-milvus-querynode-598b88956c-gc7mg      1/1     Running     0          3m3s    10.97.13.236   qa-node010.zil
liz.local   <none>           <none>
benchmark-test-7wpf9-1-milvus-querynode-598b88956c-gtwwc      1/1     Running     0          3m3s    10.97.14.244   qa-node011.zil
liz.local   <none>           <none>
benchmark-test-7wpf9-1-milvus-querynode-598b88956c-jxhkw      1/1     Running     0          3m3s    10.97.11.105   qa-node009.zil
liz.local   <none>           <none>
benchmark-test-7wpf9-1-milvus-querynode-598b88956c-rvzmt      1/1     Running     0          3m3s    10.97.17.59    qa-node014.zil
liz.local   <none>           <none>
benchmark-test-7wpf9-1-milvus-querynode-598b88956c-w99mh      1/1     Running     0          3m3s    10.97.19.84    qa-node016.zil
liz.local   <none>           <none>
benchmark-test-7wpf9-1-milvus-rootcoord-7bf9f59c75-rwvcg      1/1     Running     0          3m3s    10.97.12.40    qa-node015.zil
liz.local   <none>           <none>

**Comment by xige-16** [comment]
[float_verctor.log](https://github.com/milvus-io/milvus/files/7580881/float_verctor.log)
I did not create index to reproduce this problem, it seems that the binlog file of the vector column is wrong

**Comment by XuanYang-cn** [comment]
That's a compaction bug, will fix it soon.

**Comment by ThreadDao** [comment]
Fixed on `master-20211123-3fa5b4a`
